### PR TITLE
Change "requests" to "needs"

### PR DESCRIPTION
**Pull request by rhodesconover1**
Update all instances of "requests" to "needs" across front-end, back-end, and the db.

### CODE REVIEW

**Comment by rhodesconover1** [comment]
I finally got the authentication issue resolved! @ayrhkhan and @Tolkiennerd. 

Could you review our work when you get a chance, Matthew? Definitely open to feedback as we're new to all this!

File: Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs [hunk 1]
```diff
@@ -140,17 +140,17 @@ public async Task SetPantsSize([FromBody]KeyValuePair<int?, string> pantsSize, [
     }
     #endregion
 
-    #region Request
-    [HttpGet("requests")]
-    public async Task<Dictionary<int, string>> GetRequestsLookup([FromQuery(Name = "otid")]int outreachTeamId)
+    #region Need
+    [HttpGet("need")]
```
Comment (Tolkiennerd) [inline Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs]: This should be kept plural

File: Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs [hunk 2]
```diff
@@ -140,17 +140,17 @@ public async Task SetPantsSize([FromBody]KeyValuePair<int?, string> pantsSize, [
...
+    public async Task<Dictionary<int, string>> GetNeedLookup([FromQuery(Name = "otid")]int outreachTeamId)
```
Comment (Tolkiennerd) [inline Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs]: Keep plural (GetNeedsLookup)

File: Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx [hunk 3]
```diff
@@ -51,9 +51,9 @@ export function OutreachMapDefaultView({startingLatitude, startingLongitude}: Ma
                                         )}
                                     </div>
                                     <div >
-                                        Requests:
-                                        {housingInsecureNeighbor.requestIds.map((requestId) => 
-                                            <span>{lookups.request[requestId]}</span>
+                                        Needs:
+                                        {housingInsecureNeighbor.needIds.map((needsId) =>
```
Comment (Tolkiennerd) [inline Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx]: needsId -> needId

File: Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx [hunk 4]
```diff
@@ -51,9 +51,9 @@ export function OutreachMapDefaultView({startingLatitude, startingLongitude}: Ma
                                         )}
                                     </div>
                                     <div >
-                                        Requests:
-                                        {housingInsecureNeighbor.requestIds.map((requestId) => 
-                                            <span>{lookups.request[requestId]}</span>
+                                        Needs:
+                                        {housingInsecureNeighbor.needIds.map((needsId) => 
+                                            <span>{lookups.need[needsId]}</span>
```
Comment (Tolkiennerd) [inline Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx]: same here

**Review by Tolkiennerd — CHANGES_REQUESTED** [review]

**Comment by Tolkiennerd** [comment]
Looks good! Just a few instances to update terms to be singular/plural

File: Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs (hunk 2 above)
Comment (rhodesconover1) [reply Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs]: I believe I have addressed this.

File: Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx (hunk 3 above)
Comment (rhodesconover1) [reply Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx]: I believe I have addressed this.

File: Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx (hunk 4 above)
Comment (rhodesconover1) [reply Frontend/unhoused-outreach-ui/src/features/mapping/page-views/default.tsx]: I believe I have addressed this.

**Comment by rhodesconover1** [comment]
I believe I have referenced all the identified issues. Looking to merge.

File: Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs [hunk 5]
```diff
@@ -140,15 +140,15 @@ public async Task SetPantsSize([FromBody]KeyValuePair<int?, string> pantsSize, [
...
+    [HttpGet("needs")]
+    public async Task<Dictionary<int, string>> GetNeedsLookup([FromQuery(Name = "otid")]int outreachTeamId)
     {
         return await repository.GetNeeds(outreachTeamId);
     }
 
-    [HttpPost("need")]
-    public async Task SetNeed([FromBody]KeyValuePair<int?, string> need, [FromQuery(Name = "otid")]int outreachTeamId)
+    [HttpPost("needs")]
+    public async Task SetNeeds([FromBody]KeyValuePair<int?, string> need, [FromQuery(Name = "otid")]int outreachTeamId)
```
Comment (Tolkiennerd) [inline Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs]: This (and above line) should not have been made plural. This sets ONE need

**Review by Tolkiennerd — CHANGES_REQUESTED** [review]

File: Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs (hunk 5 above)
Comment (rhodesconover1) [reply Backend/UnhousedOutreach.Api/Controllers/LookupsController.cs]: I believe I have addressed this issue.

**Review by Tolkiennerd — APPROVED** [review]

**Pull request merged**